### ISSUE TITLE
Transaction pool and Block evaluator should use the same logic for checking transactions

### ISSUE THREAD

**Issue by ian-algorand** [algorand/go-algorand#136]
Summary: We need to consolidate two distinct pathways for checking txn validity

Reason for prioritization: fewer bugs in transaction verification, easier to reason about system

(GOAL2-431)

**Comment by zeldovich** [comment]
This turns out to be important for PR #126, so I'll do this and open another PR that should merge before #126.

**Comment by zeldovich** [comment]
See PR #150

**Comment by zeldovich** [comment]
Fixed in #150
